homeassistant/components/media_player/kef.py

```diff
@@ -0,0 +1,236 @@
+"""
+Support for interfacing HA with the KEF Wireless Speakers.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/components/media_player.kef/
+"""
+
+
+import collections
+import logging
+import time
+import json
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA, SUPPORT_SELECT_SOURCE, SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET, SUPPORT_TURN_ON, SUPPORT_VOLUME_STEP, SUPPORT_TURN_OFF,
+    MediaPlayerDevice
+)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PORT, STATE_OFF, STATE_ON
+)
+from homeassistant.helpers import config_validation as cv
+
+REQUIREMENTS = ['pykef==1.3.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+# When turing on the speaker, disable turn on for this amout of seconds
+# If the speaker has not come online in that time, turn_on can be called again.
+_TURNING_ON_TIMER_WAIT = 30.0
+
+DEFAULT_NAME = 'KEF Wireless Speaker'
+DEFAULT_PORT = 50001
+DATA_KEF = 'kef'
+
+
+SUPPORT_KEF_CLIENT_DEVICE = (
+    SUPPORT_VOLUME_SET | SUPPORT_VOLUME_STEP | SUPPORT_VOLUME_MUTE |
+    SUPPORT_SELECT_SOURCE | SUPPORT_TURN_OFF
+)
+
+CONF_TURN_ON_SERVICE = 'turn_on_service'
+CONF_TURN_ON_DATA = 'turn_on_data'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    CONF_HOST: cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_TURN_ON_SERVICE): cv.service,
+    vol.Optional(CONF_TURN_ON_DATA): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities,
+                   discovery_info=None):
+    """Set up Kef platform."""
+    import pykef
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    name = config.get(CONF_NAME)
+    speaker = pykef.KefSpeaker(host, port)
+    turn_on_service = config.get(CONF_TURN_ON_SERVICE)
```

Comment:
I don't think we should allow this. If the user wants, it's possible to use the universal media_player for this purpose.

Reply:
@MartinHjelmare Regarding this comment specifically, you say that about any part of any media player component. As LS50 is impossible to turn on without IR, and we cannot know how the user will (or will not) pass an IR command, I think this is a valid, clean, idea. It makes the component whole, and configuration simplified.

Reply:
No, we're adding a feature that the device doesn't provide. Platforms shouldn't do that. We have universal media_player and template sensors etc for that purpose, so the user can combine different integrations.

Reply:
I guess this is valid. Did not know about the universal media player. Will look at it and see if I can get the same functionality using that.